src/v/cluster/scheduling/allocation_node.cc

```diff
@@ -142,15 +142,13 @@ void allocation_node::remove_final_count(partition_allocation_domain domain) {
 }
 
 void allocation_node::update_core_count(uint32_t core_count) {
-    vassert(
-      core_count >= cpus(),
-      "decreasing node core count is not supported, current core count {} > "
-      "requested core count {}",
-      cpus(),
-      core_count);
-    auto current_cpus = cpus();
-    for (auto i = current_cpus; i < core_count; ++i) {
-        _weights.push_back(0);
+    auto current_cpus = _weights.size();
+    if (current_cpus > core_count) {
+        _weights.resize(current_cpus);
```

Comment:
should that be a resize to `core_count` ?

Reply:
also i think we have a new resize that would allow us to reduce this to one line: 
```
void resize( size_type count, const value_type& value );
```

Reply:
Good catch! Probably doesn't matter that much as the `_weights` vector is not important after flex core is enabled, but consistency is still nice to have.